sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -19,6 +21,102 @@
 from ...utils._encode import _unique
 
 
+def get_categorical_colors_and_mapping(value_list, palette="tab10"):
+    """
+    Derives a list of color and color mapping from the passed value list.
+
+    Parameters
+    ----------
+    value_list : array-like of shape (n_samples,) of str
+        Names of the categorical values by which the ICE lines are colored
+    palette: str
+        name of the matplotlib color palette
+
+    Returns
+    -------
+    list of colors
+    dictionary where the value are mapped to a color
+    """
+    if palette is None:
+        palette = "tab10"
+    color_palette = [
+        mpl.colors.to_hex(rgba_color)
+        for rgba_color in mpl.cm.get_cmap(
+            name=palette, lut=len(np.unique(value_list))
+        ).colors
+    ]
+    mapping = {v: color_palette[i] for i, v in enumerate(sorted(np.unique(value_list)))}
+    color_list = [mapping[value_list[i]] for i in range(len(value_list))]
+    return color_list, mapping
+
+
+def get_continuous_colors_and_cmap_norm(value_list, palette="viridis"):
+    """
+
+
+    Parameters
+    ----------
+    value_list : array-like of shape (n_samples,) of int or float
+        Numerical values by which the ICE lines are colored
+    palette: str
+        Name of the matplotlib color palette
+
+    Returns
+    -------
+    list of colors
+    norm: mpl.colors.Normalize
+    """
+    if palette is None:
+        palette = "viridis"
+    cmap = mpl.cm.get_cmap(palette)
+    norm = mpl.colors.Normalize(vmin=value_list.min(), vmax=value_list.max())
+    color_list = [mpl.colors.to_hex(cmap(norm(value))) for value in value_list]
+    return color_list, norm
+
+
+def get_color_list_and_legend_dict(color=None, palette=None):
+    """
+    Derives a color list and accompanying legend info from the user input.
+
+    Parameters
+    ----------
+    ice_lines_kw : dict
+            Dictionary with keywords passed to the `matplotlib.pyplot.plot` call.
+            For ICE lines in the one-way partial dependence plots.
+
+    Returns
+    -------
+    color list: List of colors
+    legend_dict: Dictionary containing info to create a legend to
+        identify individual colored lines
+    """
+    legend_dict = {}
+    value_list = np.array(color)
+    if color is None:
+        return None, None
+    # case 2: Tuple as color is not valid, but will throw matplotlib error later on
+    elif isinstance(color, tuple):
+        return None, None
+    # case 3: color argument can be a single color for all lines
+    elif np.array(color).size == 1:
+        return None, None
+    # case 1 Sequence of str -> categorical
+    elif isinstance(value_list[0], (str, bool, np.object_)):
+        color_list, mapping = get_categorical_colors_and_mapping(value_list, palette)
+        legend_dict["mapping"] = mapping
+    # case 2 Sequence of floats/int -> continuous
+    elif isinstance(value_list[0], (float, int, np.integer, np.float_)):
```

Comment:
ich glaub `np.issubdtype(value_list.dtype, np.number)` sollte laut [stackoverflow](https://stackoverflow.com/questions/29518923/numpy-asarray-how-to-check-up-that-its-result-dtype-is-numeric) auch funktionieren und wirklich alles abchecken